byway/src/main/webapp/recommendations.js

```diff
@@ -0,0 +1,254 @@
+        // by setting a timeout in delayPromise
+        await delayPromise(250);
+        const placesFound = await findPlacesWithTextSearch(request);
+        if(placesFound !== undefined) {
+          addRecommendations(request, placesFound);
+        }
+      }
+    }
+  }
+}
+
+/**
+ * Use the textSearch function from PlacesService to find the results
+ * fitting the request object passed in. Convert callback function into
+ * a chain of promises to return the result directly. If the service
+ * returns a status other than OK, reject the promise and alert the user.
+ * @param {TextSearchRequest} request object with location, radius and query fields.
+ * @return promise either with PlaceResult[] results or undefined if rejected.
+ */
+function findPlacesWithTextSearch(request) {
+  return new Promise((resolve, reject) => {
+    placesService.textSearch(request, (result, status) => {
+      if(status === "OK") {
+        resolve(result);
+      } else {
+        reject(new Error(status));
+      }
+    });
+  }).catch((error) => {
+    alertUser(error.message);
+  });
+}
+
+/**
+ * Provides a message to the user if there was an issue
+ * with their request.
+ * TODO: Make a class to indicate as a "warning" and make text red.
+ * Also, organize on front end.
+ * @param {String} msgFromService status String from placesService
+ */
+function alertUser(msgFromService) {
+  let messageContainer = document.getElementById("message-container");
+  let message = document.createElement('p');
+  if(msgFromService === "OVER_QUERY_LIMIT") {
+    message.innerText += "Showing limited results.\n"
+  } else if(msgFromService === "ZERO_RESULTS") {
+    message.innerText = "Some interests are not in the area.\n"
+  }
+  messageContainer.appendChild(message);
+}
+
+/**
+ * Places markers on the locations found from textSearch.
+ * Temporarily limit the amount of suggestions. Keep track of
+ * places found from interests around a location with savePlacesFromInterests function.
+ * TODO: Store more results and limit on UI with option to "show more"
+ * @param {TextSearchRequest} request with unique location and interest
+ * @param {PlaceResults[]} results places found with PlaceResult type.
+ */
+function addRecommendations(request, placesFound) {
+  const MAX_RECOMMENDATIONS = 1;
+  let placesLoaded = [];
+  for (place of placesFound) {
+    placeMarker(place);
+    placesLoaded.push(place);
+    if(placesLoaded.length == MAX_RECOMMENDATIONS) {
+      break;
+    }
+  }
+  savePlacesFromInterests(request, placesLoaded)
+}
+
+/**
+ * Saves the places generated from interest and location in sessionStorage as key/value pairs.
+ * @param {TextSearchRequest} request with unique location and interest
+ * @param {PlaceResults[]} placesLoaded List of places related to interest
+ */
+function savePlacesFromInterests(request, placesLoaded) {
+  sessionStorage.setItem(JSON.stringify(request), JSON.stringify(placesLoaded));
```

Comment:
I had thought the issue was only with refreshing, but I checked and the pre-loaded results after awaiting the promise hits the query limit in a new tab as well. The concern with using localStorage is having to clear the information after closing the tabs. ie if theres two tabs on the routepage, and i clear when a tab is closed, it would presumably affect the other tab still open. 

Another point is this localStorage would act like the datastore from the backend, except it would be saved on the server and not accessible on the front end via dev tools. The point of avoiding datastore was to save recommendations was to avoid saving/deleting unselected recommendations. The reason i saved on the front end temporarily was to await hitting the query limits on refresh (and new tabs), but there seem to be caveats to both approaches. 

I'm not entirely sure what the best course of action is.